src/bin/cargo/commands/install.rs

```diff
@@ -72,8 +72,13 @@ continuous integration systems.",
 }
 
 pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
+    let ws = args.workspace(config)?;
+    let host = config.rustc(Some(&ws)).map(|rustc| rustc.host).ok();
     let mut compile_opts = args.compile_options(config, CompileMode::Build)?;
     compile_opts.build_config.release = !args.is_present("debug");
+    // We override target architecture to host architecture since it may be
+    // set to some other architecture in .cargo/config.
+    compile_opts.build_config.requested_target = host;
```

Comment:
This could be just `.requesetd_target = None` I think

Reply:
Let me change that and see if that works.